### PR TITLE
Add featured projects section to home page.

### PR DESCRIPTION
**Pull request by Dhruvi16**
## Description

Adds `Featured projects` section to the home page.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: themes/vocabulary_theme/templates/blocks/featured-projects.html [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        {{ this.title }}
+      </h4>
+      <span class="level-right">
+        <a class="project-page-link" href="/contributing-code/projects/">See all available projects <i class="icon angle-right"></i></a>
+      </span>
+    </div>
+    <div class="project-list columns">
+      {% for repo in bag('repos.repos') %}
+      {% if repo.featured %}
+        <article class="card entry-post vertical project-index column is-one-third">
```
Comment (hugosolar) [inline themes/vocabulary_theme/templates/blocks/featured-projects.html:14]: Sames as blog post... let's try to keep separate the column container from the component

```
<div class="columns">
  <div class="column is-one-third">
    <article class="card entry-post vertical project-index">
      [...]
    </article>
  </div>
```

File: themes/vocabulary_theme/templates/blocks/featured-projects.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          <div class="card-content">
+            <h4 class="card-title b-header"><a href="{{ repo.url }}">{{ repo.english_name }}</a></h4>
+            <a class="button is-text tiny site-link" href="{{ repo.website }}">
+              <span>Visit site</span>
+              <i class="icon external-link"></i>
+            </a>
+            <div class="content">
+              {{ repo.description }}
+            </div>
+            <div class="labels">
```
Comment (hugosolar) [inline themes/vocabulary_theme/templates/blocks/featured-projects.html:24]: I think you need to add a `margin-bottom` on tag buttons 
Screen Shot 2020-07-06 at 4 35 25 PM

File: themes/vocabulary_theme/templates/blocks/featured-projects.html (hunk 1 above)
Comment (Dhruvi16) [reply themes/vocabulary_theme/templates/blocks/featured-projects.html:14]: Alright! And about the same height thing, I followed the designs actually, you can see it [here](https://www.figma.com/file/mttcugI1UxvCJRKE5sdpbO/Mockups?node-id=759%3A516). And if you want me to change them to same height, I just have to remove `height: 100%`

File: themes/vocabulary_theme/templates/blocks/featured-projects.html (hunk 2 above)
Comment (Dhruvi16) [reply themes/vocabulary_theme/templates/blocks/featured-projects.html:24]: I have opened an issue regarding this in vocabulary itself - [#499](https://github.com/creativecommons/vocabulary/issues/499)

File: themes/vocabulary_theme/templates/blocks/featured-projects.html (hunk 1 above)
Comment (Dhruvi16) [reply themes/vocabulary_theme/templates/blocks/featured-projects.html:14]: And regarding the third featured project - I think I should make changes in the master (through a PR). So I will raise a new PR for that. Does that make sense to you?

**Comment by Dhruvi16** [comment]
I updated the PR. For the third featured card I will raise other PR and for margin the issue is already open.

**Pull request merged**